src/corecel/cont/Span.hh

```diff
@@ -237,11 +237,10 @@ CELER_FUNCTION Span<const typename T::value_type> make_span(T const& cont)
 
 //---------------------------------------------------------------------------//
 //! Construct an array from a fixed-size span
-template<class T, std::size_t N>
-CELER_CONSTEXPR_FUNCTION Array<std::remove_cv_t<T>, N>
-make_array(Span<T, N> const& s)
+template<class T, std::size_t N, class T2 = std::remove_cv_t<T>>
+CELER_CONSTEXPR_FUNCTION Array<T2, N> make_array(Span<T, N> const& s)
```

Comment:
Wouldn't it also work without the second (somewhat confusing) template argument to do:
```suggestion
template<class T, std::size_t N>
CELER_CONSTEXPR_FUNCTION auto make_array(Span<T, N> const& s)
```
and leave the return type in place?